packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useEffect } from 'react'
+
+import {
+  Status,
+  accountSelectors,
+  developerAppSchema,
+  useAddDeveloperAppMutation
+} from '@audius/common'
+import { Button, ButtonType } from '@audius/stems'
+import { Form, Formik } from 'formik'
+import { z } from 'zod'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
+
+import { InputV2Variant } from 'components/data-entry/InputV2'
+import { TextAreaField } from 'components/form-fields/TextAreaField'
+import { TextField } from 'components/form-fields/TextField'
+import { useSelector } from 'utils/reducer'
+
+import styles from './DeveloperApps.module.css'
+import { CreateAppPageProps, CreateAppsPages } from './types'
+const { getUserId } = accountSelectors
+
+type DeveloperAppValues = z.input<typeof developerAppSchema>
+
+const messages = {
+  appNameLabel: 'App Name',
+  descriptionLabel: 'Short Description',
+  cancel: 'Cancel',
+  create: 'Create Key'
+}
+
+type CreateNewAppPageProps = CreateAppPageProps
+
+export const CreateNewAppPage = (props: CreateNewAppPageProps) => {
+  const { setPage } = props
+  const userId = useSelector(getUserId) as number
+
+  const [addDeveloperApp, result] = useAddDeveloperAppMutation()
+
+  const { status, data } = result
+
+  useEffect(() => {
+    if (status === Status.SUCCESS && data) {
+      setPage(CreateAppsPages.APP_DETAILS, data)
+    }
+  })
+
+  const handleSubmit = useCallback(
+    (values: DeveloperAppValues) => {
```

Comment:
might want to check here if user has already created 3 apps and abort if so. but we can/should also do that in the sdk...

Reply:
i think as long as we hide the create button, should be good?